### PR TITLE
Issue/2844 order detail refactor update order status

### PR DESCRIPTION
**Pull request by anitaa1990**
This PR takes care of the 10/13 step of the Order Detail refactoring #2844 .

#### Changes
- Adds the option to update order status.
- Adds the option to Issue Refund for an order.
- Adds the option to view the Refund detail screen for an order.
- Adds the option to send an sms to the billing phone, if available.
- Adds the option to call the billing phone, if available.
- Adds the option to send email to the billing email, if available.

#### Notes
- I have split this massive change into multiple small PRs in order to make reviews easier. 
- This PR is to be merged into a feature branch.
- **This PR is in draft till this [Step 9 PR](https://github.com/woocommerce/woocommerce-android/pull/2874) can be merged.**

#### Testing
- Click on an order from the Orders tab.
  - Click on the Edit icon beside the Order status.
  - Change the order status to some other status and notice the Undo snackbar is displayed.
  - Click on the Undo Snackbar and notice the status is reverted.
  - Change the status again.
  - Notice that the status is updated successfully once the SnackBar is dismissed.
- Click on an order from the Orders tab.
  - Click on the Issue Refund button.
  - Notice that once the products are refunded, the order detail is refreshed to reflect that change.
  - Notice that a new order note is added to the order after a successful refund.
- Click on an order from the Orders tab.
  - Click on the Refunds section and notice the refund detail screen is displayed.

##### Update order status
. 

##### Open Refund detail
. 

##### Issue refund
. 

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by nbradbury** [comment]
@anitaa1990 Similar to something I reported in a previous PR, if you open an order that can't be refunded, for a second it shows the "Issue refund" button and then it goes away. In addition, you can still click this button before it goes away and start the refund process.

pr

This isn't happening in `develop`.

develop

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/OrderNavigationTarget.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+sealed class OrderNavigationTarget : Event() {
+    data class ViewOrderStatusSelector(
+        val currentStatus: String,
+        val orderStatusList: Array<Order.OrderStatus>
+    ) : OrderNavigationTarget() {
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (javaClass != other?.javaClass) return false
+
+            other as ViewOrderStatusSelector
```
Comment (nbradbury) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/OrderNavigationTarget.kt:15]: **TIL**

**Comment by nbradbury** [comment]
@anitaa1990 Apart from the disappearing Refund button, this all looks great. I'll make looking at this again my first task in the morning.

**Comment by anitaa1990** [comment]
Thanks for the review @nbradbury and nice catch!! I've fixed it bcf5d23 to ensure that the issue refund button is not displayed by default.

Ready for another round 👍

**Comment by nbradbury** [comment]
> I've fixed it [bcf5d23](https://github.com/woocommerce/woocommerce-android/commit/bcf5d23695705ab5d6e7a493a7bce844f642a9ce) to ensure that the issue refund button is not displayed by default.

This fixes what I reported, but now the opposite occurs: if an order can be refunded, the Refund button doesn't appear for a second. I guess this isn't a big deal, but this doesn't happen in `develop`. What's causing the delay?

**Comment by anitaa1990** [comment]
> This fixes what I reported, but now the opposite occurs: if an order can be refunded, the Refund button doesn't appear for a second. I guess this isn't a big deal, but this doesn't happen in develop. What's causing the delay?

I think this is because all the API calls are happening sequentially in this line. I modified the code so that everything is happening in parallel in 9764f1b 👍 

Thanks for the thorough review :D Ready for another round 🤞

**Comment by nbradbury** [comment]
> I modified the code so that everything is happening in parallel in [9764f1b](https://github.com/woocommerce/woocommerce-android/commit/9764f1b232b148f8a0e9567b977522d71943c59e) 👍

This is **much** better, nice work!

**Review by nbradbury — APPROVED** [review]
:shipit:

**Pull request merged**